app.ts

```diff
@@ -0,0 +1,411 @@
+				maxRecords = maxRecords - 1;
+				let inputNumber = this.input();
+				let length = inputNumber.length;
+				let inputNumberInt = parseInt(inputNumber);
+				if (inputNumber === '') {
+					if (this.currentFromID === 0) {
+						this.currentFromID = 0;
+					}
+					let newCurrentPage = Math.ceil(this.currentFromID / maxRecords);
+					if (newCurrentPage === 1) {
+						newCurrentPage = 1;
+						this.currentFromID = 0;
+					}
+					this.currentToID = this.currentFromID + maxRecords;
+					this.currentPage = newCurrentPage;
+					let originalID = (Math.floor(maxRecords * this.currentPage) - (maxRecords - 1));
+					this.difference = this.currentFromID - originalID;
+				} else {
+					if (this.currentToID >= count) {
+						this.currentToID = (count - 1);
+					}
+					if (length > 0) {
+						let newCurrentPage = Math.ceil(inputNumberInt / maxRecords);
+						this.currentToID = newCurrentPage * maxRecords;
+						this.currentPage = newCurrentPage;
+						this.currentFromID = (this.currentPage - 1) * maxRecords + 1;
+					}
+				}
+				if (this.currentFromID === 0) {
+					pageEnd = Math.ceil(Math.floor(this.currentToID / (maxRecords - 1)) / 10) * 10;
+					if (screenHeight < 136) {
+						pageEnd = Math.ceil(Math.floor(this.currentToID / (maxRecords)) / 10) * 10;
+					}
+				} else {
+					pageEnd = Math.ceil(Math.floor(this.currentToID / (maxRecords)) / 10) * 10;
+				}
+				pageStart = pageEnd - 9;
+				this.paginationStart = pageStart;
+				this.paginationEnd = pageEnd;
+				$("#records-table-body").empty();
+				return this.showRecords(this.currentFromID, this.currentToID);
+			})
+			.then(() => {
+				return this.pageNumbers(pageStart, pageEnd);
+			})
+			.catch(error => {
+				throw new Error(`Failed when resizing the window: ${error}`);
+			});
+	}
+
+	/** When resizing the window. Timeout is put in place so that the function doesn't
+	 * take in every value returned when resizing. */
+	resize(): void {
+		if (this.resizeTimer !== null) {
+			clearTimeout(this.resizeTimer);
+		}
+		this.resizeTimer = setTimeout(() => {
+			return this.adjustDisplayedRecords()
+				.catch(error => {
+					throw error;
+				});
+		}, 250);
+	}
+
+	/** Will display when the table is empty and it's busy fetching the records. */
+	loader(): void {
+		let content = $('#records-table-body').text();
+		if (content === '') {
+			$('.results').append('<div class="loader"></div>');
+		} else {
+			$('.loader').css({ 'display': 'none' });
+		}
+	}
+
+	/** Calculate how many records should be displayed according to the screen height. */
+	recordsPerPage(): number {
+		const screenHeight = <number>($('#records-table-body').height());
+		let maxRecords = Math.floor(screenHeight / 68);
+		maxRecords = maxRecords - 1;
+		return maxRecords;
+	}
+
+	/** Retrieve the search value from the input even when it's empty. */
+	input(): string {
+		let inputNumber = <string>($('.search-input').val());
+		return inputNumber;
+	}
```

Comment:
Please change your `this.input()` to rather bring back your input string already parsed to an integer instead of doing it every time after you call it. Less code and looks clean.

If it is not a valid integer then bring back -1 or something and manage it from there.